digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala

```diff
@@ -46,74 +58,677 @@ class MunicipalityApi(val onOffLinearAssetService: OnOffLinearAssetService, val
 
   case class AssetTimeStamps(created: Modification, modified: Modification) extends TimeStamps
 
-
   before() {
     basicAuth
   }
 
-  private def extractNewLinearAssets(typeId: Int, value: JValue) = {
+  private def verifyLinearServiceToUse(typeId: Int): LinearAssetOperations = {
     typeId match {
-      case _ => value.extractOpt[Seq[NewNumericOrTextualValueAsset]].getOrElse(Nil).map(x => NewLinearAsset(x.linkId, x.startMeasure, x.endMeasure, NumericValue(x.properties.map(_.value).head.toInt), x.sideCode , x.geometryTimestamp, None))
-    }
-  }
-
-  def linearAssetsToApi(linearAssets: Seq[PersistedLinearAsset], municipalityCode: Long): Seq[Map[String, Any]] = {
-    linearAssets.map { asset =>
-      Map("id" -> asset.id,
-        "properties" -> Seq(Map("value" -> asset.value.map(_.toJson), "name" -> getAssetName(asset.typeId))),
-        "linkId" -> asset.linkId,
-        "startMeasure" -> asset.startMeasure,
-        "endMeasure" -> asset.endMeasure,
-        "sideCode" -> asset.sideCode,
-        "modifiedAt" -> asset.modifiedDateTime,
-        "createdAt" -> asset.createdDateTime,
-        "geometryTimestamp" -> asset.vvhTimeStamp,
-        "municipalityCode" -> municipalityCode,
-        "assetType" -> asset.typeId
+      case LitRoad.typeId | MassTransitLane.typeId  => onOffLinearAssetService
+      case PavedRoad.typeId => pavingService
+      case RoadWidth.typeId => roadWidthService
+      case _ => linearAssetService
+    }
+  }
+
+  private def verifyPointServiceToUse(typeId: Int): PointAssetOperations = {
+    typeId match {
+      case Obstacles.typeId  => obstacleService
+      case PedestrianCrossings.typeId   => pedestrianCrossingService
+      case RailwayCrossings.typeId  => railwayCrossingService
+      case TrafficLights.typeId  => trafficLightService
+    }
+  }
+
+  private def extractNewAssets(typeId: Int, value: JValue) = {
```

Comment:
the name should be kept as extractNewLinearAssets since it will only take care of that type, or the method below should be extractAsset